accord-core/src/test/java/accord/utils/Property.java

```diff
@@ -486,15 +486,10 @@ public <State, SystemUnderTest> void check(Commands<State, SystemUnderTest> comm
                     }
                     catch (Throwable t)
                     {
-                        try
-                        {
-                            commands.destroySut(sut, t);
-                            commands.destroyState(state, t);
-                        }
-                        catch (Throwable t2)
-                        {
-                            t.addSuppressed(t2);
-                        }
+                        State finalState = state;
+                        safeHandle(t, () -> commands.onFailure(finalState, sut, maybeRewriteHistory(history, historyTiming), t));
```

Comment:
`onFailure` is mostly there for when i do tests in a loop i want to get a summary of the history i am looking at.  For example while working on CASSANDRA-20156 i want to find a shorter history, so getting a summary of whats going on helps me see if i should ignore the failing seed.

Also, if `destorySut` failed `destoryState` wouldn't get called, so this logic is now properly handled